### DISCUSSION TITLE
Could you please add support for FXML Language?

### DISCUSSION THREAD

**Discussion by saifali-dev** [boyter/scc#227] [Q&A] [answered]
### [FXML Language](https://en.wikipedia.org/wiki/FXML)
It is an XML-Based language for JavaFX UI. 
You guys have already done awesome work. And I think it would be easy for you to add one more XML-like language.

Just to count FXML code, I have to use cloc and then scc to calculate the rest the rest.
It is really annoying for me to switch between both tools just because of one language.

I hope you will understand my request.

Thanks!

**Comment by boyter** [comment, marked as answer]
That shouldn't be too hard. Do you have a language spec for is it? Or is it just an XML variant? Would counting it as XML be sufficient (but with a different name?)

https://docs.oracle.com/javafx/2/api/javafx/fxml/doc-files/introduction_to_fxml.html

based on the above it just looks like XML to me.

Does it have a specific file extension?

In short I need to populate the following,


```
"FXML": {
    "extensions": [
      "fxml"
    ],
    "line_comment": [],
    "multi_line": [
      [
        "<!--",
        "-->"
      ]
    ],
    "quotes": [
      {
        "end": "\"",
        "start": "\""
      }
    ]
  },
```

**Reply by boyter** [reply]
Poking around seems the above might just be it. Have committed the change to master. If you want a build you can build from there to get this support right now. Otherwise there will be a new release sometime in the next month as soon as Go 1.16 is released :)

**Reply by boyter** [reply]
https://github.com/boyter/scc/issues/228

**Reply by saifali-dev** [reply]
> Would counting it as XML be sufficient (but with a different name?)

Yes, Sufficient and with **FXML** as the language name.

> based on the above it just looks like XML to me.

You are **right!** It is **exactly same as XML** but with **fxml extension**.

> Does it have a specific file extension?

Only **fxml**.


### Your commit is perfect for this, Thank you super much!

## Jazzak Allah

**Reply by boyter** [reply]
Great! Will make it into the next release then, otherwise feel free to use master which should be stable enough :)

**Reply by saifali-dev** [reply]
Sure, and please consider updating [language count](https://github.com/boyter/scc#languages) to 245 **:)**

**Reply by boyter** [reply]
Hah I should automate that so its always in sync.
